src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/factory/generator/flow/SingleTableIngressRuleGenerator.java

```diff
@@ -0,0 +1,118 @@
+import org.openkilda.match.Match;
+import org.openkilda.model.FlowEncapsulationType;
+import org.openkilda.model.FlowEndpoint;
+import org.openkilda.model.MeterId;
+import org.openkilda.model.Switch;
+import org.openkilda.model.SwitchFeature;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import lombok.experimental.SuperBuilder;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+@SuperBuilder
+public class SingleTableIngressRuleGenerator extends IngressRuleGenerator {
+
+    @Override
+    public List<Command> generateCommands(Switch sw) {
+        List<Command> result = new ArrayList<>();
+        Command meterCommand = buildMeter(flowPath.getMeterId());
+        result.add(meterCommand);
+
+        FlowSideAdapter flowSide = FlowSideAdapter.makeIngressAdapter(flowPath);
+        FlowCommand command = buildFlowCommand(sw, flowSide);
+        result.add(command);
+        if (command != null) {
+            if (meterCommand != null) {
+                command.getDependsOn().add(meterCommand.getUuid());
+            }
+        }
+
+        return result;
+    }
+
+    private Command buildMeter(MeterId meterId) {
+        // todo fix meter
+        return MeterCommand.builder().build();
+    }
+
+    private FlowCommand buildFlowCommand(Switch sw, FlowSideAdapter flowSide) {
+        FlowEndpoint endpoint = flowSide.getEndpoint();
+        Collection<Match> match = Lists.newArrayList(
+                FieldExactMatch.builder().field(Field.IN_PORT).value(endpoint.getPortNumber()).build());
+        if (isDefaultFlow(flowSide)) {
+            match.add(FieldExactMatch.builder().field(Field.VLAN_VID).value(endpoint.getOuterVlanId()).build());
+        }
+        List<Action> actions = new ArrayList<>();
+        if (!isDefaultFlow(flowSide)) {
+            actions.add(new PopVlanAction());
+        }
+        actions.addAll(buildTransitEncapsulationActions());
+        Instructions instructions = buildInstructions(flowPath.getMeterId().getValue(), sw, actions);
+
+        FlowCommandBuilder<?, ?> builder = FlowCommand.builder()
+                .uuid(UUID.randomUUID().toString())
+                .commandType(CommandType.FLOW)
+                .switchId(endpoint.getSwitchId())
+                .cookie(flowPath.getCookie().getValue())
+                .tableNumber(Constants.TableNumber.INPUT_TABLE_ID)
+                .priority(isDefaultFlow(flowSide) ? Constants.Priority.DEFAULT_FLOW_PRIORITY
+                        : Constants.Priority.FLOW_PRIORITY)
+                .match(match)
+                .instructions(instructions);
+
+        if (sw.getFeatures().contains(SwitchFeature.RESET_COUNTS_FLAG)) {
+            builder.flags(Sets.newHashSet(FlowFlag.RESET_COUNTERS));
+        }
+        return builder.build();
+    }
+
+    private boolean isDefaultFlow(FlowSideAdapter flowSide) {
```

Comment:
... default it terrible term, at least for me. Especially when you call the service rules as default rules. Name it `isFullPortFlow`.

Reply:
Renamed.